sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -488,7 +489,6 @@ def create_item(
         request_options = build_options(kwargs)
         response_hook = kwargs.pop('response_hook', None)
 
-        request_options["disableAutomaticIdGeneration"] = True
```

Comment:
As I mentioned above - we probably need to keep this here to ensure consistent behaviour for existing customers, allowing the parameter overwrite:
```python
request_options["disableAutomaticIdGeneration"] = kwargs.pop('disable_auto_id_gen', True)
```

Reply:
that makes sense - I was thinking this wouldn't affect users since it requires them to use an ID already, but I can see people relying on getting an error when they do not pass an ID

Reply:
Thanks @zfoster - just to clarify, when you say "requires them to use an ID already", whereabouts is that?

Reply:
``` 
request_options["disableAutomaticIdGeneration"] = True
````

having this hardcoded means you have to supply an ID in the args when creating items

Reply:
if i'm reading it right

Reply:
Interesting - in that case what is the keyword argument people would need to have been supplying? And is that what our tests are doing? (sorry I should have dug deeper into this earlier)

Reply:
we didn't have a kwarg for it afaict. honestly, I'm not sure why this is the way it is, it's unique to this sdk. Our tests do supply IDs and from this we can assume our users do too, however I can flesh out why I think your comment about ensuring consistent behavior is still important:

We may have users who generate IDs or create them programmatically, and rely on the fact that we don't ever automatically generate them. The way this is set up now, we keep this consistent and never generate IDs unless they explicitly pass the kwarg in (same behavior as before)